### PR TITLE
Integrate OAuth with Okta. 

### PR DESCRIPTION
**Pull request by gregcharles**
Switch between OpenAM and Okta is based on configuration. For now, this configuration remains pointed to OpenAM, so it can be merged to develop without impact. 

Please add reviewers as needed. I just made my best guess. :)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -96,6 +96,9 @@ dependencies {
 
     compile 'com.lmax:disruptor:3.3.7'
 
+    compile group: 'com.okta.jwt', name: 'okta-jwt-verifier', version: '0.4.0'
+    compile group: 'com.okta.jwt', name: 'okta-jwt-verifier-impl', version: '0.4.0'
```
Comment (malaffoon) [inline build.gradle]: inconsistent format from other entries in this file

File: src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java [hunk 2]
```diff
@@ -13,29 +18,32 @@
...
+import java.time.Duration;
 
 @Configuration
 @EnableZuulProxy
 @EnableResourceServer
 @EnableConfigurationProperties({ApiGatewayProperties.class})
 public class ApiGatewayConfiguration extends WebSecurityConfigurerAdapter {
 
+    private static final String OKTA_TOKEN_PROVIDER = "okta";
     private final ApiGatewayProperties apiGatewayProperties;
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java]: not part of these changes but ... this seems like an anti-pattern?

File: src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java [hunk 3]
```diff
@@ -75,4 +89,20 @@ public void handleError(ClientHttpResponse response) throws IOException {
 
         return new OpenAmTokenServices(this.apiGatewayProperties, tokenConverter, restTemplate);
     }
-}
\ No newline at end of file
+
+    private ResourceServerTokenServices oktaTokenServices() {
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java]: nit: this isn't an annotated bean function but i still think it is a general practice to return the specific type, i.e. `private OktaTokenServices()`.

File: build.gradle (hunk 1 above)
Comment (gregcharles) [reply build.gradle]: Changed to match style

File: src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java (hunk 3 above)
Comment (gregcharles) [reply src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java]: Changed to return specific types.

File: src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java (hunk 2 above)
Comment (gregcharles) [reply src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java]: As you point out, the state was already part of this bean, and I was trying to change as little as possible. Should I refactor it though? Or create a tech debt ticket in TIMS?

Comment (millerts) [reply src/main/java/org/opentestsystem/ap/apigateway/ApiGatewayConfiguration.java]: Greg, if you're willing please go ahead and make the change.  If not, please let me know.  Do not create tickets in the TIMS project.

**Comment by gregcharles** [comment]
@millerts I merged as is, but I'll circle back to refactoring ApiGatewayConfiguration.java if I can fit it in. Either way, I'll let you know the status.

**Pull request merged**